pandas/core/categorical.py

```diff
@@ -374,11 +374,31 @@ def itemsize(self):
 
     def reshape(self, new_shape, *args, **kwargs):
         """
-        An ndarray-compatible method that returns
-        `self` because categorical instances cannot
-        actually be reshaped.
+        DEPRECATED: calling this method will raise an error in a
+        future release.
+
+        An ndarray-compatible method that returns `self` because
+        `Categorical` instances cannot actually be reshaped.
+
+        Parameters
+        ----------
+        new_shape : int or tuple of ints
+            A 1-D array of integers that correspond to the new
+            shape of the `Categorical`. The total size indicated
+            by `new_shape` must be the same as that of the original
+            shape of the `Categorical`. One shape dimension can be
+            negative. In this case, the value is inferred from the
+            length of the array and the remaining dimensions.
```

Comment:
As this is really a `numpy` thing, I'm just going to point to `np.reshape` documentation (we're deprecating this function after-all, so I don't feel too bad for tying it to `numpy`).